### PR TITLE
Proper Output Consolidation In Wallet Lib

### PR DESCRIPTION
**Pull request by lzpap**
# Description of change

 - Wallet lib (and cli wallet) can consolidate any number of outputs sitting on wallet addresses.
 - In the unlikely scenario that even after the first try, there are still more, than `MaxInputCount` outputs on wallet addresses, just run the command again.
 - Regular wallet functions fail with too many inputs, and notify users to consolidate funds.

### CODE REVIEW

File: client/wallet/wallet.go [hunk 1]
```diff
@@ -181,59 +187,68 @@ func (wallet *Wallet) ConsolidateFunds(options ...consolidateoptions.Consolidate
...
-	}
-	if consolidateOptions.WaitForConfirmation {
-		err = wallet.WaitForTxConfirmation(tx.ID())
+		wallet.markOutputsAndAddressesSpent(consumedOutputs)
+		err = wallet.connector.SendTransaction(tx)
+		if err != nil {
+			return nil, err
+		}
+		if consolidateOptions.WaitForConfirmation {
+			err = wallet.WaitForTxConfirmation(tx.ID())
```
Comment (capossele) [inline client/wallet/wallet.go:248]:
```suggestion
			err = wallet.WaitForTxConfirmation(tx.ID())
			if err != nil {
				return nil, err
			}
```

Comment (capossele) [reply client/wallet/wallet.go:248]: Double check that this is what should happen in case of error.

File: client/wallet/output.go [hunk 2]
```diff
@@ -167,3 +167,43 @@ func (o OutputsByAddressAndOutputID) ToLedgerStateOutputs() ledgerstate.Outputs
...
+
+// SplitIntoChunksOfMaxInputCount splits the mapping into chunks that contain at most ledgerstate.MaxInputCount outputs.
+func (o OutputsByAddressAndOutputID) SplitIntoChunksOfMaxInputCount() []OutputsByAddressAndOutputID {
+	outputCount := o.OutputCount()
+	if outputCount <= ledgerstate.MaxInputCount {
+		// there is no need to split
+		return []OutputsByAddressAndOutputID{o}
+	}
+	result := make([]OutputsByAddressAndOutputID, outputCount/ledgerstate.MaxInputCount+1)
+	for i, _ := range result {
```
Comment (capossele) [inline client/wallet/output.go:190]:
```suggestion
	for i := range result {
```

File: client/wallet/wallet.go [hunk 3]
```diff
@@ -181,59 +187,68 @@ func (wallet *Wallet) ConsolidateFunds(options ...consolidateoptions.Consolidate
...
-	if err != nil {
-		return nil, err
-	}
-	if !ok {
-		return nil, errors.Errorf("created transaction is invalid: %s", tx.String())
-	}
+		// check tx validity (balances, unlock blocks)
+		ok, err := checkBalancesAndUnlocks(inputsAsOutputsInOrder, tx)
+		if err != nil {
+			return nil, err
```
Comment (capossele) [inline client/wallet/wallet.go:236]:
```suggestion
		ok, e := checkBalancesAndUnlocks(inputsAsOutputsInOrder, tx)
		if e != nil {
			return nil, e
```

**Review by capossele — CHANGES_REQUESTED** [review]

File: client/wallet/wallet.go (hunk 1 above)
Comment (lzpap) [reply client/wallet/wallet.go:249]: I've refactored it.

**Review by capossele — APPROVED** [review]

**Pull request merged**